transformers4rec/torch/features/embedding.py

```diff
@@ -497,3 +497,15 @@ def forward(self, input_numeric):
         soft_one_hot_embeddings = (weights.unsqueeze(-1) * self.embedding_table.weight).sum(-2)
 
         return soft_one_hot_embeddings
+
+
+class PretrainedEmbeddingsInitializer(torch.nn.Module):
+    def __init__(self, weight_matrix, trainable=False, **kwargs):
+        super().__init__(**kwargs)
+        self.weight_matrix = torch.tensor(weight_matrix)
```

Comment:
Thanks @sararb . I am now assigning the weight matrix to the CPU, so that it does not use GPU memory. In the `forward()` those weights are copied (`x.copy_()`) to the device where the embedding table is.